virtualization/windowscontainers/manage-containers/container-networking.md

```diff
@@ -211,6 +199,45 @@ In order to see which containers are connected to a specific network and the IPs
 C:\> docker network inspect <network name>
 ```
 
+### Specify the Name of a Network to the HNS Service
+
+If you are creating a network, you can specify the name that it is given by the HNS service using the option, `-o com.docker.network.windowsshim.networkname=<network name>` to the `docker network create` command. For instance, you might use the following command to create a transparent network with a name that is specified to the HNS service:
+
+```none
+C:\> docker network create -d transparent -o com.docker.network.windowsshim.networkname=MyTransparentNetwork MyTransparentNetwork
+```
+Ordinarily, when you create a container network using `docker network create`, the network name that you provide is used by the Docker service but not by the HNS service. As a result, you will see the name of your network when you view your network using the `docker network ls` command (as documented above), but you will not see your chosen name if you view your network or network switch using the `Get-VMSwitch` or `Get-ContainerNetwork` Windows PowerShell commands (instead you will see a name that was automatically generated by the HNS service).
+
+### Bind a Network to a Specific Network Interface
+
+To bind a network (attached through the Hyper-V virtual switch) to a specific network interface, use the option, `-o com.docker.network.windowsshim.interface=<Interface>` to the `docker network create` command. For instance, you might use the following command to create a transparent network which is attached to the "Ethernet 2" network interface:
+
+```none
+C:\> docker network create -d transparent -o com.docker.network.windowsshim.interface="Ethernet 2" TransparentNet2
+```
+
+> Note: The value for *com.docker.network.windowsshim.interface* is the network adapter's *Name*, which can be found with:
+
+```none
+PS C:\> Get-NetAdapter
+```
+
+### Set the VLAN ID for a Network
+
+To set a VLAN ID for a network, use the option, `-o com.docker.network.windowsshim.vlanid=<VLAN ID>` to the `docker network create` command. For instance, you might use the following command to create a transparent network with a VLAN ID of 11:
+
+```none
+C:\> docker network create -d transparent -o com.docker.network.windowsshim.vlanid=11 MyTransparentNetwork
+```
+
```

Comment:
Containers attached to a network with VLAN specified will have the VLAN isolation set on the VM Network adapter of the container endpoint. Show output of Get-VMNetworkAdapter (first)  Get-VMNetworkAdapterVlan <Name>. Also mention the fact that Host Network Adapter should be in Trunk mode. The Management vNIC may be on a separate VLAN (for host management) Get-VMNetworkAdapterVlan -ManagementOs